compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1751,6 +1752,11 @@ impl<'tcx> TyCtxt<'tcx> {
         self.mk_ty(Array(ty, ty::Const::from_usize(self, n)))
     }
 
+    #[inline]
+    pub fn mk_array_const(self, ty: Ty<'tcx>, ct: Const<'tcx>) -> Ty<'tcx> {
```

Comment:
I'm not sure if I love this name -- `mk_array` and rename the existing one to `mk_array_from_usize`?

Reply:
The existing `mk_arrray` is used a lot more than `mk_array_const`. What about changing the latter to `mk_array_from_const`?

Reply:
That works too! The `mk_array_const` name by itself makes it sound like this should be returning a `Const<'tcx>` or a valtree or something.

Reply:
hmmmmmmmm.... actually, that still kinda feels like it's a "make a const" function name. But I have no better names in mind, so oh well.

Reply:
`mk_array_from_const` sounds to me like it makes an array.

Reply:
`mk_array_with_ty_const_len` or `mk_array_with_const_len` maybe

Reply:
I'll use `mk_array_with_const_len`.